File: Section Reports/Data/XML/C#/XML/CustomersOrders.cs
namespace GrapeCity.ActiveReports.Samples.XML
{
	public partial class CustomersOrders : GrapeCity.ActiveReports.SectionReport
	{
		public CustomersOrders()
		{
			InitializeComponent();
		}
		/// <summary>
		/// ReportStart Event
		/// Use this event to initialize any objects or variables needed while running a report. 
		/// Also use this event to set any Subreport control objects to a new instance 
		/// of the report assigned to the Subreport control.
		/// </summary>
		private void CustomersOrders_ReportStart(object sender, System.EventArgs eArgs)
		{
			//Create the subrepot in this event instead of  
			//setting it in the Detail Format Event everytime. 
			srptOrders.Report = new Orders();
		}
	}
}

File: Section Reports/Data/XML/VB.NET/XML/CustomersOrders.vb
Public Class CustomersOrders
	Inherits GrapeCity.ActiveReports.SectionReport
	Public Sub New()
		MyBase.New()
		InitializeComponent()
	End Sub
	' Format Event
	' ActiveReports raises this event after the data is loaded and bound to the controls contained in a section, 
	' but before the section is rendered to a page.The Format event is the only event in which you can change
	' the section's height. Use this section to set or change the properties of any controls or the section itself.
	' Also use the Format event to pass information, such as an SQL String, to a Subreport.
	Private Sub Detail_Format_1(ByVal sender As System.Object, ByVal e As EventArgs) Handles Detail.Format
		'Set subreport object per line.
		srptOrders.Report = New Orders()
	End Sub
End Class
